### PR TITLE
Default mpi log format requires fancyrecord

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: lib/vsc/utils/fancylogger.py [hunk 1]
```diff
@@ -341,15 +343,22 @@ def getLogger(name=None, fname=False, clsname=False, fancyrecord=None):
...
+        # it's easier to explain if we do it this way
+        fancyrecord = FANCYLOG_FANCYRECORD
+
     if name:
         nameparts.append(name)
-    elif fancyrecord is None or fancyrecord:  # only be fancy if fancyrecord is True or no name is given
+    elif fancyrecord is None:  # only be fancy if fancyrecord is True or no name is given
         fancyrecord = True
+
     fancyrecord = bool(fancyrecord)  # make sure fancyrecord is a nice bool, not None
```
Comment (boegel) [inline lib/vsc/utils/fancylogger.py]: drop this? seems to me fancyrecord can never be a non-bool anymore (unless something else than `None` or `True` or `False` is passed as a function argument)

File: test/fancylogger.py [hunk 2]
```diff
@@ -369,6 +369,46 @@ def test_normal_logging(self):
         fancylogger.logToScreen(enable=False, handler=handler)
         sys.stderr = _stderr
 
+    def test_fancyrecord(self):
+        """
+        Test fancyrecord usage
+        """
+        logger=fancylogger.getLogger()
+        self.assertTrue(logger.fancyrecord == True)
```
Comment (boegel) [inline test/fancylogger.py]: use `assertEqual` when comparing values, which will lead to clear test failure messages

File: test/fancylogger.py [hunk 3]
```diff
@@ -369,6 +369,46 @@ def test_normal_logging(self):
...
+
+        logger=fancylogger.getLogger('myname')
+        self.assertTrue(logger.fancyrecord == True)
+
+        logger=fancylogger.getLogger('myname', fancyrecord=False)
+        self.assertTrue(logger.fancyrecord == False)
+
+        fancylogger.FANCYLOG_FANCYRECORD=orig
```
Comment (boegel) [inline test/fancylogger.py]: ` = ` (spaces)

File: lib/vsc/utils/fancylogger.py (hunk 1 above)
Comment (stdweird) [inline lib/vsc/utils/fancylogger.py]: so leave it there in case someone does?

File: test/fancylogger.py [hunk 4]
```diff
@@ -369,6 +369,46 @@ def test_normal_logging(self):
         fancylogger.logToScreen(enable=False, handler=handler)
         sys.stderr = _stderr
 
+    def test_fancyrecord(self):
+        """
+        Test fancyrecord usage
+        """
+        logger=fancylogger.getLogger()
```
Comment (boegel) [inline test/fancylogger.py]: ` = ` (spaces)

File: test/fancylogger.py [hunk 5]
```diff
@@ -369,6 +369,46 @@ def test_normal_logging(self):
...
+        logger=fancylogger.getLogger(fancyrecord=False)
+        self.assertTrue(logger.fancyrecord == False)
+
+        logger=fancylogger.getLogger('myname')
+        self.assertTrue(logger.fancyrecord == False)
+
+        logger=fancylogger.getLogger('myname', fancyrecord=True)
+        self.assertTrue(logger.fancyrecord == True)
+
+        orig=fancylogger.FANCYLOG_FANCYRECORD
```
Comment (boegel) [inline test/fancylogger.py]: ` = ` (spaces)

and better use this on top of the test method?

File: test/fancylogger.py [hunk 6]
```diff
@@ -369,6 +369,46 @@ def test_normal_logging(self):
...
+        fancylogger.FANCYLOG_FANCYRECORD=False
```
Comment (boegel) [inline test/fancylogger.py]: ` = ` (spaces)

File: test/fancylogger.py [hunk 7]
```diff
@@ -369,6 +369,46 @@ def test_normal_logging(self):
...
+
+        logger=fancylogger.getLogger()
```
Comment (boegel) [inline test/fancylogger.py]: ` = ` (spaces), same below

File: test/fancylogger.py [hunk 8]
```diff
@@ -378,7 +418,6 @@ def tearDown(self):
         fancylogger.FancyLogger.RAISE_EXCEPTION_CLASS = self.orig_raise_exception_class
         fancylogger.FancyLogger.RAISE_EXCEPTION_LOG_METHOD = self.orig_raise_exception_method
 
-
```
Comment (boegel) [inline test/fancylogger.py]: please keep this?

**Comment by boegel** [comment]
@stdweird: only minor remarks, thanks for tackling the tests too!

File: lib/vsc/utils/fancylogger.py (hunk 1 above)
Comment (boegel) [inline lib/vsc/utils/fancylogger.py]: yeah, may be safest, I guess...

**Comment by stdweird** [comment]
@boegel done

File: lib/vsc/utils/fancylogger.py (hunk 1 above)
Comment (boegel) [inline lib/vsc/utils/fancylogger.py]: maybe add a test case that checks that things still work when passing a non-boolean value then?

```python
logger = fancylogger.getLogger('myname', fancyrecord='yes')
self.assertEqual(logger.fancyrecord, True)

logger = fancylogger.getLogger('myname', fancyrecord=0)
self.assertEqual(logger.fancyrecord, False)
```

**Comment by boegel** [comment]
alrighty, going in!

**Pull request merged**